src/Mod/Arch/archobjects/base_rebar.py

```diff
@@ -0,0 +1,256 @@
+                    "App::Property",
+                    "The mark number of this rebar shape"
+                )
+            )
+        # Length
+        if "Length" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "Length",
+                "Rebar Shape",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "The length of this rebar shape"
+                )
+            )
+            obj.setEditorMode("Length", 1)
+
+    def onDocumentRestored(
+        self,
+        obj
+    ):
+        ArchComponent.Component.onDocumentRestored(self, obj)
+        self.setProperties(obj)
+
+    def execute(
+        self,
+        obj
+    ):
+        if self.clone(obj):
+            return
+        if not obj.Base:
+            FreeCAD.Console.PrintError(
+                "No Base, return without a rebar shape for {}.\n"
+                .format(obj.Name)
+            )
+            return
+        if not obj.Base.Shape:
+            FreeCAD.Console.PrintError(
+                "No Shape in Base, return without a rebar shape for {}.\n"
+                .format(obj.Name)
+            )
+            return
+        if not obj.Base.Shape.Edges:
+            FreeCAD.Console.PrintError(
+                "No Edges in Shape of Base, "
+                "return without a rebar shape for {}.\n"
+                .format(obj.Name)
+            )
+            return
+        if obj.Base.Shape.Faces:
+            FreeCAD.Console.PrintError(
+                "Faces in Shape of Base, "
+                "return without a rebar shape for {}.\n"
+                .format(obj.Name)
+            )
+            return
+        if not obj.Diameter.Value:
+            FreeCAD.Console.PrintError(
+                "No diameter value, return without a rebar shape for {}.\n"
+                .format(obj.Name)
+            )
+            return
+        if not obj.MarkNumber:
+            FreeCAD.Console.PrintError(
+                "No mark number, return without a rebar shape for {}.\n"
+                .format(obj.Name)
+            )
+            return
+
+        # corner cases:
+        #    compound from more Wires
+        #    compound without Wires but with multiple Edges
+        # Does they make sense? If yes handle them.
+        # Does it makes sense to handle Shapes with Faces or even Solids?
+        if not obj.Base.Shape.Wires and len(obj.Base.Shape.Edges) == 1:
+            wire = Part.Wire(obj.Base.Shape.Edges[0])
+        else:
+            wire = obj.Base.Shape.Wires[0]
+        edge = wire.Edges[0]
+        bpoint = edge.Vertexes[0].Point
+        bvec = edge.tangentAt(edge.FirstParameter)
+        if not bpoint:
+            return
+        # all tests ok!
+
+        if hasattr(obj, "Rounding"):
+            if obj.Rounding:
+                radius = obj.Rounding * obj.Diameter.Value
+                from DraftGeomUtils import filletWire
```

Comment:
`Part` is heavy and it is imported at the top of the file, therefore, all other modules should be imported at the top of the part as well. In particular `DraftGeomUtils` also imports `Part` internally, so we don't save time by delaying its import.

I suggest using the `lazy_loader` module introduced in the Path workbench to deal with the issue of delaying imports. See `src/3rdParty/lazy_loader`, and the forum thread, [New python library for deferred loading](https://forum.freecadweb.org/viewtopic.php?f=10&t=45298).

I will test it in Draft as well.